### PR TITLE
[Hub Generated] Review request for Search to add version preview/2022-09-01-preview

### PR DESCRIPTION
**Pull request by rishikaparashar**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/dev-maps-Search-2022-09-01-preview...main/).
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*engagement record*](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/271/Azure-REST-API-Stewardship?anchor=rest-api-stewardship-process).

* Link to engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [x] a public preview
- [ ] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous Open API document (swagger) if applicable, and the root paths that have been updated. </sup>
* Design Document:
* Previous Open API Doc:
* Updated paths:

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azapi) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azapi).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

File: specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json [hunk 1]
```diff
@@ -0,0 +1,966 @@
...
+        "modelAsString": true,
+        "values": [
+          {
+            "value": "FeatureCollection"
+          }
+        ]
+      }
+    },
+    "Boundary": {
+      "description": "`GeoJSON GeocodingFeature` object that describe the boundaries of a geographical area. Geometry of the feature is described with `GeoJSON GeometryCollection`.\n\nPlease note for the Boundary search Polygon API, the GeometryCollection Polygon and MultiPolygon are the only geometry objects expected in the main response for now.",
```
Comment (jsedlak-microsoft) [inline specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json:915]: Please note, the service typically returns a GeometryCollection with Polygon or Mulitpolygon sub-types.

File: specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json [hunk 2]
```diff
@@ -0,0 +1,966 @@
...
+          }
+        ]
+      },
+      "x-ms-parameter-location": "method"
+    }
+  },
+  "paths": {
+    "/geocode": {
+      "get": {
+        "description": "**Geocoding**\n\n**Applies to:** see pricing [tiers](https://aka.ms/AzureMapsPricingTier).\n\nIn many cases, the complete search service might be too much, for instance if you are only interested in traditional geocoding. Search can also be accessed for address look up exclusively. The geocoding is performed by hitting the geocoding endpoint with just the address or partial address in question. The geocoding search index will be queried for everything above the street level data. No POIs will be returned. Note that the geocoder is very tolerant of typos and incomplete addresses. It will also handle everything from exact street addresses or street or intersections as well as higher level geographies such as city centers, counties, states etc.\n>[!Important]\n>By using this feature, you agree to the preview legal terms. See the [Preview Supplemental Terms](https://azure.microsoft.com/en-us/support/legal/preview-supplemental-terms/) for additional details.",
```
Comment (jsedlak-microsoft) [inline specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json:225]: Maybe write out 'point of interest (POIs)'

Comment (jsedlak-microsoft) [reply specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json:225]: POIs is a common enough term without spelling it out though.

**Comment by rishikaparashar** [comment]
Please review this PR by Igor for all comments and suggestions.
https://github.com/Azure/azure-rest-api-specs/pull/20245

File: specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json [hunk 3]
```diff
@@ -0,0 +1,966 @@
...
+      },
+      "collectionFormat": "csv",
+      "minItems": 4,
+      "maxItems": 4,
+      "x-ms-parameter-location": "method"
+    },
+    "View": {
+      "name": "view",
+      "in": "query",
+      "description": "A string that an [ISO 3166-1 Alpha-2 region/country code](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2). This will alter Geopolitical disputed borders and labels to align with the specified user region.\n\nPlease refer to [Supported Views](https://aka.ms/AzureMapsLocalizationViews) for details and to see the available Views.",
```
Comment (pbrasil) [inline specification/maps/data-plane/Search/preview/2022-09-01-preview/search.json:104]: We should callout that the default value is "auto".

**Review by pbrasil — APPROVED** [review]

**Comment by rishikaparashar** [comment]
@weidongxu-microsoft it was reviewed. Please refer https://github.com/Azure/azure-rest-api-specs/pull/20245 for notes and recommendations from the API review board.

**Review by joshchen1 — APPROVED** [review]
This PR is based on https://github.com/Azure/azure-rest-api-specs/pull/20245. It only changes one of the parameters from 'location' to 'coordinate' in /search/polygon and it looks good.

**Review by dubiety — APPROVED** [review]

**Comment by weidongxu-microsoft** [comment]
> This PR is based on #20245. It only changes one of the parameters from 'location' to 'coordinate' in /search/polygon and it looks good.

approved in https://github.com/Azure/azure-rest-api-specs/pull/20245 by Mark and Mike.

We should be able to merge this without further approval. Ping me or assignee when ready.

**Pull request merged**